odd-platform-ui/src/components/Management/DataSourcesList/DataSourceItem/DataSourceItemToken/DataSourceItemToken.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { isBefore, subSeconds } from 'date-fns';
+import {
+  DataSource,
+  DataSourceApiUpdateDataSourceRequest,
+} from 'generated-sources';
+import * as S from 'components/Management/DataSourcesList/DataSourceItem/DataSourceItemToken/DataSourceItemTokenStyles';
+import ConfirmationDialog from 'components/shared/ConfirmationDialog/ConfirmationDialog';
+import { Typography } from '@mui/material';
+import AppButton from 'components/shared/AppButton/AppButton';
+import CopyButton from 'components/shared/CopyButton/CopyButton';
+
+interface DataSourceItemProps {
+  dataSource: DataSource;
+  updateDataSource: (
+    params: DataSourceApiUpdateDataSourceRequest
+  ) => Promise<DataSource>;
+}
+
+const DataSourceItemToken: React.FC<DataSourceItemProps> = ({
+  dataSource,
+  updateDataSource,
+}) => {
+  const [isHidden, setIsHidden] = useState(
+    isBefore(dataSource.token.updatedAt, subSeconds(new Date(), 1))
+  );
```

Comment:
Please, use hooks annotation via `React.useState`, like you did it with React.useEffect below.